torch/distributed/algorithms/model_averaging/utils.py

```diff
@@ -0,0 +1,20 @@
+import torch
+import torch.distributed as dist
+
+
+def average_parameters(
+    model: torch.nn.Module, process_group: dist.ProcessGroup = dist.group.WORLD
+):
+    """
+    Averages all the parameters of a given model.
+    For allreduce efficiency, all the parameters are flattened into a contiguous buffer.
+    Thus, it requires extra memory of the same size as the model’s parameters.
+    """
+    flat_params = torch.cat([p.data.view(-1) for p in model.parameters()])
+    flat_params /= float(dist.get_world_size(group=process_group))
```

Comment:
nit: I don't think we need the cast to float here?

Reply:
Yeah, removed this unnecessary explicit casting.